lua/lua/flasher_helper.lua

```diff
@@ -0,0 +1,676 @@
+
+	-- open the plugin
+	tPlugin, strMsg = getPlugin(strPluginName, strPluginType, atPluginOptions)
+
+	if tPlugin ~= nil then
+		fOk, strMsg = pcall(tPlugin.Connect, tPlugin)
+		if not fOk then
+			strMsg = strMsg or "Failed to open connection"
+		else
+			local iChiptype = tPlugin:GetChiptyp()
+			local strChiptypName = tPlugin:GetChiptypName(iChiptype)
+			local ulWdgBaseAddr = atChiptyp2WatchdogBase[iChiptype]
+
+			if ulWdgBaseAddr == nil then
+				-- unknown chip type or not supported
+				strMsg = string.format("The reset_netx command is not supported on %s (%d)", strChiptypName, iChiptype)
+
+			elseif iChiptype == romloader.ROMLOADER_CHIPTYP_NETIOLA or
+				iChiptype == romloader.ROMLOADER_CHIPTYP_NETIOLB then
+				-- Watchdog reset on netIOL
+
+				local ulAddr_wdg_sys_cfg            = ulWdgBaseAddr + 0
+				local ulAddr_wdg_sys_cmd            = ulWdgBaseAddr + 4
+				local ulAddr_wdg_sys_cnt_upper_rld  = ulWdgBaseAddr + 8
+				local ulAddr_wdg_sys_cnt_lower_rld  = ulWdgBaseAddr + 12
+				local ulPwd = 0x3fa * 4
+				local ulVal
+
+				-- disable watchdog
+				tPlugin:write_data16(ulAddr_wdg_sys_cfg, ulPwd)
+
+				-- check if it is disabled
+				ulVal = tPlugin:read_data16(ulAddr_wdg_sys_cfg)
+				ulVal = ulVal % 2
+				--ulVal = bit.band(ulVal, 1)
+				if ulVal ~= 0 then
+					print("Warning: cannot disable watchdog on netIOL")
+				end
+
+				-- todo: what values for prescaler/counter?
+				tPlugin:write_data16(ulAddr_wdg_sys_cnt_upper_rld, 0x07ff)
+				tPlugin:write_data16(ulAddr_wdg_sys_cnt_lower_rld, 0xffff)
+
+				-- enable watchdog
+				tPlugin:write_data16(ulAddr_wdg_sys_cfg, ulPwd + 1)
+
+				-- trigger watchdog
+				tPlugin:write_data16(ulAddr_wdg_sys_cmd, 0x72b4)
+				tPlugin:write_data16(ulAddr_wdg_sys_cmd, 0xde80)
+				tPlugin:write_data16(ulAddr_wdg_sys_cmd, 0xd281)
+
+				print ("The netX should reset after one second.")
+				fOk = true
+
+			else
+				-- watchdog reset on other netX types
+				local ulAddr_WdgCtrl       = ulWdgBaseAddr + 0
+				local ulAddr_WdgIrqTimeout = ulWdgBaseAddr + 8
+				local ulAddr_WdgResTimeout = ulWdgBaseAddr + 12
+				local ulVal
+
+				-- Set write enable for the timeout regs
+				ulVal = tPlugin:read_data32(ulAddr_WdgCtrl)
+				ulVal = ulVal + 0x80000000
+				tPlugin:write_data32(ulAddr_WdgCtrl, ulVal)
+
+				-- IRQ after 0.9 seconds (9000 * 100µs, not handled)
+				tPlugin:write_data32(ulAddr_WdgIrqTimeout, 9000)
+				-- reset 0.1 seconds later
+				tPlugin:write_data32(ulAddr_WdgResTimeout, 1000)
+
+				-- Trigger the watchdog once to start it
+				ulVal = tPlugin:read_data32(ulAddr_WdgCtrl)
+				ulVal = ulVal + 0x10000000
+				tPlugin:write_data32(ulAddr_WdgCtrl, ulVal)
+
+				print ("The netX should reset after one second.")
+				fOk = true
+			end
+
+			tPlugin:Disconnect()
+			tPlugin = nil
+
+			-- Wait 1 second (actually between 1 and 2 seconds)
+			if (fOk == true) then
+				sleep_s(1)
+			end
+		end
+	end
+
+	return fOk, strMsg
+end
+
+function switch_endian(ulValue)
```

Comment:
check this function for correctness. Also, where is it used and is it necessary?